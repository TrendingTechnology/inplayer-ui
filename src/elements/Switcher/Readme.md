Toggle is a input component for flipswitch. You can use it directly like this:

```jsx static
import { Label, Switcher } from '@inplayer-org/inplayer-ui';

class Page extends React.Component {
   state = {
    checked: false,
  };

  onToggleChange = checked => {
    this.setState({
      checked,
    });
  };

  return <Switcher checked={value} id="toggle1" label="swipe" onChange={this.onToggleChange} />
}
```

### Demo

```jsx
<Switcher
  disabled={false}
  checked={true}
  id="toggle1"
  label="swipe"
  onChange={console.log}
/>
```
