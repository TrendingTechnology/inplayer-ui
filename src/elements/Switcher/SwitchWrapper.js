import styled from 'styled-components';
import uiColors from 'utils/uiColors';
import colors from 'config/colors';

const SwitchWrapper = styled.span`
  display: inline-flex;
  align-items: center;
  justify-content: center;

  .react-toggle-thumb {
    transition: all 0.5s cubic-bezier(0.23, 1, 0.32, 1) 0ms;
    position: absolute;
    top: -1px;
    left: 0;
    width: 22px;
    height: 22px;
    border-radius: 50%;
    border: 1px solid ${colors.gray};
    background: ${colors.white};
    box-sizing: border-box;
  }

  .react-toggle-track {
    width: 34px;
    height: 18px;
    padding: 0;
    border-radius: 30px;
    background-color: ${colors.lightGray};
    transition: all 0.2s ease;
    border: 1px solid ${colors.gray};
  }

  .react-toggle {
    touch-action: pan-x;
    display: inline-block;
    position: relative;
    cursor: pointer;
    background-color: transparent;
    border: 0;
    padding: 0;
    user-select: none;
  }

  .react-toggle-screenreader-only {
    border: 0;
    clip: rect(0 0 0 0);
    height: 1px;
    margin: -1px;
    overflow: hidden;
    padding: 0;
    position: absolute;
    width: 1px;
  }

  .react-toggle--disabled {
    background: #00000005 !important;
    cursor: auto;
  }

  .react-toggle--disabled .react-toggle-track {
    background: #00000005 !important;
    border: 1px solid ${colors.gray} !important;
  }

  .react-toggle--disabled .react-toggle-thumb {
    background: #f0f3f6;
    border: 1px solid ${colors.gray};
  }

  .react-toggle--checked .react-toggle-track {
    border: 1px solid ${uiColors('primary.main')};
    background: ${uiColors('primary.main')};
  }

  .react-toggle--checked .react-toggle-thumb {
    left: 14px;
  }

  &:hover .react-toggle-track {
    background: rgba(0, 170, 230, 0.06);
    border: 1px solid #00aae6;
  }

  &:hover .react-toggle-thumb {
    border: 1px solid #00aae6;
  }

  .react-toggle:hover:not(.react-toggle--disabled) .react-toggle-track,
  .react-toggle--checked:hover:not(.react-toggle--disabled) .react-toggle-track {
    background: ${uiColors('primary.main')};
    border: 1px solid ${uiColors('primary.main')};
  }

  .react-toggle:hover:not(.react-toggle--disabled) .react-toggle-thumb {
    border: 1px solid ${uiColors('primary.main')};
  }

  .react-toggle-track-check {
    position: absolute;
    width: 14px;
    height: 10px;
    top: 0;
    bottom: 0;
    margin-top: auto;
    margin-bottom: auto;
    line-height: 0;
    left: 8px;
    opacity: 0;
    transition: opacity 0.25s ease;
  }

  .react-toggle--checked .react-toggle-track-check {
    opacity: 1;
    transition: opacity 0.25s ease;
  }

  .react-toggle-track-x {
    position: absolute;
    width: 10px;
    height: 10px;
    top: 0;
    bottom: 0;
    margin-top: auto;
    margin-bottom: auto;
    line-height: 0;
    right: 10px;
    opacity: 1;
    transition: opacity 0.25s ease;
  }

  .react-toggle--checked .react-toggle-track-x {
    opacity: 0;
  }
`;

/** @component */
export default SwitchWrapper;
