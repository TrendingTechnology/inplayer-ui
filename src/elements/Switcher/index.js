// @flow
import React from 'react';
import Toggle from 'react-toggle';

import SwitchWrapper from './SwitchWrapper';
import Label from '../Label';

type Props = {
  checked: boolean,
  disabled: boolean,
  id: string,
  label: string,
  onChange: (checked: boolean) => any,
};

const Switcher = ({ id, checked, label, disabled, onChange }: Props) => {
  const onToggleChange = (e: SyntheticInputEvent<*>) => {
    if (e.target) {
      onChange(e.target.checked);
    }
  };
  return (
    <SwitchWrapper>
      <Toggle
        id={id}
        disabled={disabled}
        defaultChecked={checked}
        onChange={onToggleChange}
        icons={false}
      />
      <Label htmlFor={id}>{label}</Label>
    </SwitchWrapper>
  );
};

/** @component */
export default Switcher;
