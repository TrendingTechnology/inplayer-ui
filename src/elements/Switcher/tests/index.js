import { generateSnapshots } from 'helpers';

import Switcher from '../index';

generateSnapshots(Switcher);
