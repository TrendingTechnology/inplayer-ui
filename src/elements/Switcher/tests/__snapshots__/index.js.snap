// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button renders correctly 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background: #d7dde5;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #a0a3aa;
  font-weight: 600;
  border: 1px solid #d7dde5;
  padding: 10px 20px 11px;
  border-radius: 3px;
  height: 1.125rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  outline: none;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  width: 1.125rem;
  cursor: pointer;
  vertical-align: middle;
}

.c0:hover,
.c0:focus {
  -webkit-transform: scale(1.2);
  -ms-transform: scale(1.2);
  transform: scale(1.2);
}

.c0:hover span,
.c0:focus span {
  color: #00aae6;
}

<button
  className="c0"
/>
`;

exports[`Toggle renders correctly 1`] = `
.c0 {
  width: 34px;
  background: #f0f3f6;
  border: 1px solid #d7dde5;
  border-radius: 50px;
  padding: 9px;
  margin: 12px 0;
  -webkit-transition: all ease 500ms;
  transition: all ease 500ms;
  overflow: visible;
}

.c0:checked {
  border: 1px solid #00aae6;
  background: #00aae6;
}

.c0:hover,
.c0:checked:hover {
  background: #00aae60f;
  border: 1px solid #00aae6;
}

.c0::after {
  content: '';
  display: block;
  position: absolute;
  border: 1px solid #d7dde5;
  top: -2px;
  left: -2px;
  width: 20px;
  height: 20px;
  z-index: 2;
  background: #fff;
  border-radius: 50px;
  -webkit-transition: all ease 300ms;
  transition: all ease 300ms;
}

.c0:hover::after {
  border: 1px solid #00aae6;
}

.c0:checked::after {
  left: 13px;
}

<input
  className="c0"
/>
`;

exports[`undefined renders correctly 1`] = `
.c0 {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 .react-toggle-thumb {
  -webkit-transition: all 0.5s cubic-bezier(0.23,1,0.32,1) 0ms;
  transition: all 0.5s cubic-bezier(0.23,1,0.32,1) 0ms;
  position: absolute;
  top: -1px;
  left: 0;
  width: 22px;
  height: 22px;
  border-radius: 50%;
  border: 1px solid #d7dde5;
  background: #fff;
  box-sizing: border-box;
}

.c0 .react-toggle-track {
  width: 34px;
  height: 18px;
  padding: 0;
  border-radius: 30px;
  background-color: #f0f3f6;
  -webkit-transition: all 0.2s ease;
  transition: all 0.2s ease;
  border: 1px solid #d7dde5;
}

.c0 .react-toggle {
  touch-action: pan-x;
  display: inline-block;
  position: relative;
  cursor: pointer;
  background-color: transparent;
  border: 0;
  padding: 0;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c0 .react-toggle-screenreader-only {
  border: 0;
  -webkit-clip: rect(0 0 0 0);
  clip: rect(0 0 0 0);
  height: 1px;
  margin: -1px;
  overflow: hidden;
  padding: 0;
  position: absolute;
  width: 1px;
}

.c0 .react-toggle--disabled {
  background: #00000005 !important;
  cursor: auto;
}

.c0 .react-toggle--disabled .react-toggle-track {
  background: #00000005 !important;
  border: 1px solid #d7dde5 !important;
}

.c0 .react-toggle--disabled .react-toggle-thumb {
  background: #f0f3f6;
  border: 1px solid #d7dde5;
}

.c0 .react-toggle--checked .react-toggle-track {
  border: 1px solid #00aae6;
  background: #00aae6;
}

.c0 .react-toggle--checked .react-toggle-thumb {
  left: 14px;
}

.c0:hover .react-toggle-track {
  background: rgba(0,170,230,0.06);
  border: 1px solid #00aae6;
}

.c0:hover .react-toggle-thumb {
  border: 1px solid #00aae6;
}

.c0 .react-toggle:hover:not(.react-toggle--disabled) .react-toggle-track,
.c0 .react-toggle--checked:hover:not(.react-toggle--disabled) .react-toggle-track {
  background: #00aae60f;
  border: 1px solid #00aae6;
}

.c0 .react-toggle:hover:not(.react-toggle--disabled) .react-toggle-thumb {
  border: 1px solid #00aae6;
}

.c0 .react-toggle-track-check {
  position: absolute;
  width: 14px;
  height: 10px;
  top: 0;
  bottom: 0;
  margin-top: auto;
  margin-bottom: auto;
  line-height: 0;
  left: 8px;
  opacity: 0;
  -webkit-transition: opacity 0.25s ease;
  transition: opacity 0.25s ease;
}

.c0 .react-toggle--checked .react-toggle-track-check {
  opacity: 1;
  -webkit-transition: opacity 0.25s ease;
  transition: opacity 0.25s ease;
}

.c0 .react-toggle-track-x {
  position: absolute;
  width: 10px;
  height: 10px;
  top: 0;
  bottom: 0;
  margin-top: auto;
  margin-bottom: auto;
  line-height: 0;
  right: 10px;
  opacity: 1;
  -webkit-transition: opacity 0.25s ease;
  transition: opacity 0.25s ease;
}

.c0 .react-toggle--checked .react-toggle-track-x {
  opacity: 0;
}

.c1 {
  -webkit-text-decoration: none;
  text-decoration: none;
  font-weight: 300;
  cursor: pointer;
  vertical-align: middle;
  color: #44494e;
  -webkit-transition: all ease 300ms;
  transition: all ease 300ms;
  line-height: 20px;
  margin: 0 20px 0 -14px;
  left: 24px;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
}

.c1:hover {
  color: #a0a3aa;
}

.c1:disabled {
  color: #c8ccd4;
  cursor: default;
}

<span
  className="c0"
>
  <div
    className="react-toggle"
    onClick={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="react-toggle-track"
    >
      <div
        className="react-toggle-track-check"
      />
      <div
        className="react-toggle-track-x"
      />
    </div>
    <div
      className="react-toggle-thumb"
    />
    <input
      className="react-toggle-screenreader-only"
      onBlur={[Function]}
      onChange={[Function]}
      onFocus={[Function]}
      type="checkbox"
    />
  </div>
  <label
    className="c1"
  />
</span>
`;
